assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -93,25 +108,24 @@ function MediaProvider({ children }) {
   } = useConfig();
 
   const resetWithFetch = useCallback(() => {
+    // eslint-disable-next-line no-shadow
+    const { mediaType, pagingNum, searchTerm } = stateRef.current;
```

Comment:
We already have an effect where we call `fetchMedia` on state changes. I think the best option would be to reuse the same effect and just add an additional trigger to it.

Reply:
resetWithFetch is used to update the media when the media picker is closed:
https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/library/panes/media/mediaPane.js#L152

Reply:
What would the additional trigger look like?

Reply:
Yes. Let's do in a separate pr. But to clarify, the additional trigger could simply be:

```
const [resetter, setResetter] = useState(1);
const resetMedia = useCallback(() => setResetter(value => value + 1), []);

useEffect(() => fetchMedia(...), [...allTheCurrentDeps, resetter]);

return {
  ...
  resetMedia,
  ...
}
```

A bit crude, but helps reduce dependencies if critical.

Reply:
https://github.com/google/web-stories-wp/issues/2152